qooxdoo-kit/Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM node:10-alpine
+
+LABEL maintainer=oetiker
+
+# adding these in separate layers since
+RUN apk add --no-cache su-exec git python2 tini
```

Comment:
Can install in ``--virtual`` instead of in ``world`` and uninstall at the end. This will reduce even further the container size :-)

See example e.g. [.BUILD-DEPS](https://github.com/pcrespov/itis-dockerfiles/blob/is5/alpine-pandas/python-with-pandas/3.6-alpine/Dockerfile#L5) in this ``Dockerfile``

Reply:
After some tests last week, it does not reduce the container size unless it is done in the same RUN command because the previous layers cannot be altered.

Reply:
Something like 
```dockerfile
RUN set -e; \
    apk add --no-cache --virtual .build-deps \
    gcc \
    libc-dev \
    linux-headers \
    git \
    ; \
    pip install -r src/requirements.txt; \
    apk del .build-deps;
```

Reply:
nice!

Reply:
integrated

Reply:
> Can install in `--virtual` instead of in `world` and uninstall at the end. This will reduce even further the container size :-)
> 
> See example e.g. [.BUILD-DEPS](https://github.com/pcrespov/itis-dockerfiles/blob/is5/alpine-pandas/python-with-pandas/3.6-alpine/Dockerfile#L5) in this `Dockerfile`

note that the example will NOT make the image smaller, because a layer gets generated for each line in the setup. so you have to remove the stuff in the same RUN line you are adding it, to get the benefit of the smaller image.